server/statistics/topn.go

```diff
@@ -54,61 +58,46 @@ func (tn *TopN) Len() int {
 	return tn.ttlLst.Len()
 }
 
-// GetTopNMin returns the min item in top N.
-func (tn *TopN) GetTopNMin() TopNItem {
+// GetTopNMin returns the min item in top N of the `k`th dimension.
+func (tn *TopN) GetTopNMin(k int) TopNItem {
 	tn.rw.RLock()
 	defer tn.rw.RUnlock()
-	return tn.topn.Top()
+	return tn.topns[k].GetTopNMin()
 }
 
-// GetAllTopN returns the top N items.
-func (tn *TopN) GetAllTopN() []TopNItem {
+// GetAllTopN returns the top N items of the `k`th dimension.
+func (tn *TopN) GetAllTopN(k int) []TopNItem {
 	tn.rw.RLock()
 	defer tn.rw.RUnlock()
-	return tn.topn.GetAll()
+	return tn.topns[k].GetAllTopN()
 }
 
 // GetAll returns all items.
 func (tn *TopN) GetAll() []TopNItem {
 	tn.rw.RLock()
 	defer tn.rw.RUnlock()
-	topn := tn.topn.GetAll()
-	return append(topn, tn.rest.GetAll()...)
+	return tn.topns[0].GetAll()
```

Comment:
No need `GetAll(k int)`?

Reply:
No, all items in `TopN` will be in all dimensions.

Reply:
Why only `topns[0]` in here?

Reply:
Any dimension contains all items, so `tn.topns[0].GetAll() == tn.topns[i].GetAll()` for `i = 1, 2, 3, ...`.